src/Shared/FileMatcher.cs

```diff
@@ -1486,12 +1479,18 @@ private static int LastIndexOfDirectoryOrRecursiveSequence(string str, int start
             out bool isLegalFileSpec)
         {
             GetFileSpecInfo(filespec,
-                out _, out _, out _,
-                out string matchFileExpression, out needsRecursion, out isLegalFileSpec);
-            
-            regexFileMatch = isLegalFileSpec
-                ? new Regex(matchFileExpression, DefaultRegexOptions)
-                : null;
+                out string fixedDirectoryPart, out string wildcardDirectoryPart, out string filenamePart,
+                out needsRecursion, out isLegalFileSpec);
+
+            if (isLegalFileSpec)
+            {
+                string matchFileExpression = RegularExpressionFromFileSpec(fixedDirectoryPart, wildcardDirectoryPart, filenamePart);
```

Comment:
RegularExpressionFromFileSpec starts by checking whether IsLegalFileSpec(...). It's private and hardly used. Could we pull that out here so we don't check that twice?

Reply:
Actually, I think you can just delete it. It seems to be checked before every call.

Reply:
The check is already gone, were you maybe looking at an older version of the PR. Thanks!